frontend/app/[team]/apps/[app]/environments/[environment]/[[...path]]/page.tsx

```diff
@@ -798,10 +968,66 @@ export default function EnvironmentPath({
               <FolderBreadcrumbLinks path={params.path} />
             </div>
             {unsavedChanges && (
+              <div onClick={handleChangesReveal}>
               <Alert variant="warning" icon={true} size="sm">
-                You have undeployed changes to this environment.
+                <span className="mr-[-0.5rem]" onClick={handleChangesReveal}>
+                  You have undeployed changes to this environment.
+                </span>
+                <GenericDialog
+                  buttonVariant=""
+                  title="Undeployed changes"
+                  onClose={() => setIsChangesModalOpen(false)}
+                  buttonContent={<FaInfoCircle />}
+                >
+                  <div className="flex flex-col space-y-2">
+                    <h1 className='text-zinc-500 mb-5'>The following changes are pending deployment</h1>
+                    {Object.entries(allChanges).map(([id, change]) => (
+                      <div key={id} className="flex flex-row items-center">
+                        {change.type === "Added" && (
+                          <div className='flex flex-col mb-6'>
+                            <div className='flex flex-row items-center space-x-2'>
+                              <GoDotFill className="text-emerald-400" />
+                              <p className="text-emerald-400">
+                                Created 
+                              </p>
+                              <p className='dark:text-white text-black font-mono'>{change.secretName}</p>
+                            </div>
+                            <DisplayChanges change={change} />
+                          </div>
+                        )}
+                        {change.type === "Modified" && (
+                          <div className="flex flex-col mb-6">
+                            <div className="flex items-center space-x-2">
+                              <div className='flex flex-row space-x-2 items-center'>
+                                <GoDotFill className="text-yellow-500" />
+                                <p className="text-amber-400 ">
+                                  Updated 
+                                </p>
+                                <p className='dark:text-white text-black font-mono'>{change.secretName}</p>
+                              </div>
+                            </div>
+                            <DisplayChanges change={change} />
+                          </div>
+                        )}
+                      </div>
+                    ))}
+                    {secretsToDelete?.map(secretId => {
+                      const deletedSecret = serverSecrets.find(secret => secret.id === secretId);
+                      return (
+                        <div className="flex items-center space-x-2 mb-6">
```

Comment:
```suggestion
                        <div key={secretId} className="flex items-center space-x-2 mb-6">
```

Add a key for elements rendered in a list